### DISCUSSION TITLE
Stronghold Migration Error - Help please

### DISCUSSION THREAD

**Discussion by andythewise** [iotaledger/firefly#8566] [Help & Support]
Am receiving: "Stronghold migration error: failed to decrypt snapshot: incorrect password or corrupt data". And have no idea of how to proceed. Help would be greatly appreciated. Best. A
PS; Using Firefly version 2.0.3

**Comment by begonaalvarezd** [comment]
The message suggests that the password you are using to decrypt your stronghold is incorrect. If you dont remember your password you can always restore your profile using your 24 words. 

The latest version is v2.0.11, you can try upgrading

**Reply by andythewise** [reply]
Hi and thank. Password is correct. On desktop there is no option to restore the profile. Have restored wallet on ipad successfully with seed phrase but there are virtually no funds shown. Also upgraded desktop to version 2.0.11 which gives me the same old error message. It cannot be that the wallet sinply stops working, can it? Any other lead you might think of? Best. A

**Reply by begonaalvarezd** [reply]
Firefly always has an option to restore your profile. Just create a new profile and go through the process of restoring, instead of creating a brand new one (see screenshot, second option). 
![image](https://github.com/iotaledger/firefly/assets/3624944/fab56371-7e3c-48a5-ac04-b808482de583)


Firefly is only supported in desktop by the way, we had mobile support in the past but it got deprecated in July 2023 and the old mobile version is not compatible with the latest protocol. You should use our desktop version.

If you manage to restore your profile, remember to run the balance finder, you can access it from settings > advanced, run it couple of times to scan your addresses
